vcr/issuer/leia_store.go

```diff
@@ -39,36 +41,35 @@ type leiaIssuerStore struct {
 
 // NewLeiaIssuerStore creates a new instance of leiaIssuerStore which implements the Store interface.
 func NewLeiaIssuerStore(dbPath string) (Store, error) {
-	store, err := leia.NewStore(dbPath, false)
+	store, err := leia.NewStore(dbPath)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create leiaIssuerStore: %w", err)
 	}
-	collection := store.Collection("issuedCredentials")
+	collection := store.JSONCollection("issuedCredentials")
 	newLeiaStore := &leiaIssuerStore{
 		issuedCredentials: collection,
 		store:             store,
 	}
-	if err := newLeiaStore.createIndices(); err != nil {
+	if err := newLeiaStore.createIndices(collection); err != nil {
 		return nil, err
 	}
 	return newLeiaStore, nil
 }
 
 func (s leiaIssuerStore) StoreCredential(vc vc.VerifiableCredential) error {
 	vcAsBytes, _ := json.Marshal(vc)
-	doc := leia.DocumentFromBytes(vcAsBytes)
-	return s.issuedCredentials.Add([]leia.Document{doc})
+	return s.issuedCredentials.Add([]leia.Document{vcAsBytes})
 }
 
 func (s leiaIssuerStore) SearchCredential(jsonLDContext ssi.URI, credentialType ssi.URI, issuer did.DID, subject *ssi.URI) ([]vc.VerifiableCredential, error) {
-	query := leia.New(leia.Eq("issuer", issuer.String())).
-		And(leia.Eq("type", credentialType.String())).
-		And(leia.Eq("@context", jsonLDContext.String()))
+	query := leia.New(leia.Eq(leia.NewJSONPath("issuer"), leia.MustParseScalar(issuer.String()))).
```

Comment:
`MustParseScalar` will never fail here? (because they're strings?). Not wanting to cause a panic here at some point in future.

Reply:
it can only fail if the given argument is not a string, bool or float64